File: ARL- Apriori/aprioriARL.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jun  7 23:43:34 2019

@author: Venkatesh Ravichandran

Association Rule Learing - Apriori algorithm

"""

#import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#read csv data
df = pd.read_csv('Market_Basket_Optimisation.csv', header = None)

#convert the dataset format to a list of lists which is accepted as input to apriori algorithm
transactions = []
for i in range(0,7501):
    transactions.append([str(df.values[i,j]) for j in range(0,20)]) 

#apriori algorithm to deduct the rules
from apyori import apriori
rules = apriori(transactions, min_support = 0.003, min_confidence = 0.2, min_lift = 3, min_length = 2)

#Visualisation of rules
results = list(rules)

File: ARL- Apriori/aprioriARL.R

# read csv data
df <- read.csv('Market_Basket_Optimisation.csv', header = FALSE)
#install.packages('arules')
library(arules)
#For Apriori - We convert the dataset to sparse matrix(Matrix which contains 0s and 1s where 0s are more)
df <- read.transactions('Market_Basket_Optimisation.csv', sep = ',', rm.duplicates = TRUE)
summary(df)
#to plot the frequency of all the items purchased
itemFrequencyPlot(df, topN = 100)

#Apriori algorithm
rules = apriori(df, parameter = list(support = 0.004,confidence = 0.2))

#displaying the rules
inspect(sort(rules, by = 'lift')[1:10])
